platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/stats/ReconnectMapMetrics.java

```diff
@@ -0,0 +1,167 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.merkle.synchronization.stats;
+
+import com.swirlds.metrics.api.Counter;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+
+/**
+ * An implementation of ReconnectMapStats that emits all the stats as Counter metrics.
+ */
+public class ReconnectMapMetrics implements ReconnectMapStats {
+
+    private static final String RECONNECT_MAP_CATEGORY = "ReconnectMap";
+
+    private final ReconnectMapStats aggregateStats;
+
+    private final Counter transfersFromTeacher;
+    private final Counter transfersFromLearner;
+
+    private final Counter internalHash;
+    private final Counter internalCleanHash;
+    private final Counter internalData;
+    private final Counter internalCleanData;
+
+    private final Counter leafHash;
+    private final Counter leafCleanHash;
+    private final Counter leafData;
+    private final Counter leafCleanData;
+
+    /**
+     * Create an instance of ReconnectMapMetrics.
+     * @param metrics a non-null Metrics object
+     * @param label an optional label, e.g. a VirtualMap name or similar, may be null.
+     *     If specified, the label is added to the names of the metrics.
+     * @param aggregateStats an optional aggregateStats object to which all ReconnectMapStats calls will delegate
+     *     in addition to emitting metrics for this object, may be null.
+     */
+    public ReconnectMapMetrics(
+            @NonNull final Metrics metrics,
+            @Nullable final String label,
+            @Nullable final ReconnectMapStats aggregateStats) {
+        Objects.requireNonNull(metrics, "metrics must not be null");
+        this.aggregateStats = aggregateStats;
+
+        this.transfersFromTeacher = metrics.getOrCreate(
+                new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("transfersFromTeacher", label))
+                        .withDescription("number of transfers from teacher to learner"));
+        this.transfersFromLearner = metrics.getOrCreate(
+                new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("transfersFromLearner", label))
+                        .withDescription("number of transfers from learner to teacher"));
+
+        this.internalHash =
+                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalHash", label))
```

Comment:
Here and below: should these names be plural rather than single? As written now, full metric name will be `internalHash_accounts_Total`. `internalHashes` look more natural to me

Reply:
Sounds good. Renamed the "hash" ones. Please note that "data" is already plural.